### ISSUE TITLE
HexDebug - endianness

### ISSUE THREAD

**Issue by vlopes11** [dusk-network/dusk-bytes#5]
Some structs (e.g. BlsScalar) serialize their bytes as little endian.

This means the printed chars are inverted. This can be misleading, sometimes.

Proposal:
The procedural macro should consider as default the little endian encoding, and should have a flag to indicate if the serialization is performed as big endian.

Example:

`println!("{:x}", BlsScalar::one());`

Will print: "0100000000000000000000000000000000000000000000000000000000000000"
Should print: "0000000000000000000000000000000000000000000000000000000000000001"

**Comment by ZER0** [comment]
To be clear, this is intended. The purpose of `HexDebug` is not display "the value" of the entire structure (if the structure represent a value) but the bytes representation in hexadecimal. So if ` BlsScalar::one()` will prints `0100000000000000000000000000000000000000000000000000000000000000` we know that the first byte of the structure is`01`. It makes no assumption on in which order the structure decide to encode, if in little endian or big endian. So it's up to the structure designer.

We might want to discuss further on this topic, but generically speaking this functionality wants to display the bytes representation "as is", as the `to_bytes` returns the value.

You can compare as a dump of memory address.

Basically what is important is what is the position of the bytes in most of the case when we debug (e.g. when we send over gRPC or FFI some datastructure) to compare that the first byte we receive is actually the one we expected; and not that  `"0100000000000000000000000000000000000000000000000000000000000000"` meaning "one" and it's just encoded in little endian, because this logic is related to the structure internally and not generic.

But I will add the label `question` so we'll take this topic during the Q/A.

**Comment by ZER0** [comment]
We discussed the issue during the stand up and we all agreed to keep as is in order to avoid unnecessary over complication.
If in the future we'll have a scenario were endianess makes sense for `HexDebug` will discuss a possible solution.

Closing then as wontfix.
